gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+DeclareOperation( "SetVertexCoordiantes3DNC", [IsDenseList, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup GetVertexCoordinates3D
+#! @Description
+#! Extract the 3D-coordinates from the print record of the vertex in position <A>i</A> of <K>Vertices</K> from the corresponding <A>surface</A>. If the format of the 3D-coordinates (3D-coordinates have to be a list of 3 entries of floats) is not correct, then an error is shown. This can happen, if the NC version is used to store the 3D-coordinates.
+#! The NC-version does not check the coordinate format saved in the print record.
+#! @Returns the 3D-coordinates of the <A>i</A>th vertex.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetVertexCoordiantes3D", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @Arguments i, printRecord
+DeclareOperation( "GetVertexCoordiantes3DNC", [IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup DrawSurfaceToJavaScript
+#! @Description
+#! These functions animate the <A>surface</A> as a html file <A>filename</A> in JavaScript into the working directory. The animation can be watched and opened with any browser.
+#! An introduction to the use of this method (along with two examples)
+#! can be found at the start of section
+#! <Ref Sect="Section_LabelIntroductionAndQuickStartAnimating"/>.
+#!
+#! * If the given <A>fileName</A> does not end in <E>.html</E> the ending
+#!   <E>.html</E> will be added to it.
+#! * The given file will be overwritten without asking if it already exists.
+#!   If you don't have permission to write in that file, this method will
+#!   throw an error.
+#! * The particulars of the drawing are determined by the
+#!   given <A>printRecord</A>.
+#! * The <A>printRecord</A> will be modified and returned by this method.
+#!   It contains the data to recreate the drawing of the surface.
+#!
+#! To use these functions it is necessary to set the 3D-coordinates (3D-coordinates are defined as a list with 3 entries of floats) of the vertices of the surface (see <K>SetVertexCoordinates3D</K>).
+#!
+#! The function <K>DrawSurfaceToJavaScript</K> always calculates the parameters which are needed corresponding to the current positions of the vertices. Therefore the <A>printRecord</A> will be changed if some parameters are not set to the actual vertice positions.
+#!
+#! The function <K>DrawSurfaceToJavaScriptCalculate</K> has an extra parameter <A>calculate</A> which is a bool (true or false value). If you want to avoid that the parameters are always recalculated you can use the function <K>DrawSurfaceToJavaScriptCalculate</K> and set <A>calculate = false</A>. If, based on the configuration in the print record, calculations are possible and necessary, they will be done. This means that, if there are no parameters set for the edges and the inner circles the corresponding functions <K>CalculateParametersOfEdges</K> and <K>CalculateParametersOfInnerCircle</K> will be called.
```

Comment:
(<K>true</K> or <K/>false</K>)
To avoid permanent recalculation of derived locations, like for edges, (which should not be an issue during regular usage), you can use ...

I would replace your long explanation with this:
.. <A>false</A>. Calculations are only performed at the first call of the method that needs to draw the derived objects.